src/main/java/graphql/language/Directive.java

```diff
@@ -25,14 +30,38 @@ public Directive(String name, List<Argument> arguments) {
         return arguments;
     }
 
+    public Map<String, Argument> getArgumentsMap() {
+        // the spec says that args MUST be unique within context
+        return toNameMap(arguments, Argument::getName);
+    }
+
+    public Argument getArgument(String argumentName) {
+        return getArgumentsMap().get(argumentName);
+    }
+
     public String getName() {
         return name;
     }
 
-    public void setName(String name) {
-        this.name = name;
+    /*
```

Comment:
Can we separate this helper from the actual class? For examples in a class `Directives` following JDK Conventions.